File: 7-kyu-kata/shorter-concat-reverse-longer/python/shorter_concat_reverse_longer_test.py
# 7 kyu - shorter concat [reverse longer]  [ ID: 54557d61126a00423b000a45  (shorter-concat-reverse-longer) ]
# URL: https://www.codewars.com/kata/54557d61126a00423b000a45
# Category: REFERENCE  |  Tags: STRINGS | FUNDAMENTALS
# ******************************************************************************
import random, string


def random_string():
    l = random.randint(0, 10)
    output = ""
    for i in range(l):
        output += string.ascii_lowercase[random.randint(0, 25)]
    return output


def test_shorter_reverse_longer(a, b):
    return a + b[::-1] + a if len(b) > len(a) else b + a[::-1] + b


test.assert_equals(shorter_reverse_longer("first", "abcde"), "abcdetsrifabcde")
test.assert_equals(shorter_reverse_longer("hello", "bau"), "bauollehbau")
test.assert_equals(shorter_reverse_longer("abcde", "fghi"), "fghiedcbafghi")
for i in range(100):
    a, b = random_string(), random_string()
    test.assert_equals(shorter_reverse_longer(a, b), test_shorter_reverse_longer(a, b))

File: 7-kyu-kata/shorter-concat-reverse-longer/javascript/shorterConcatReverseLonger.Test.js
// 7 kyu - shorter concat [reverse longer]  [ ID: 54557d61126a00423b000a45  (shorter-concat-reverse-longer) ]
// URL: https://www.codewars.com/kata/54557d61126a00423b000a45
// Category: REFERENCE  |  Tags: STRINGS | FUNDAMENTALS
// *****************************************************************************
Test.assertEquals(shorter_reverse_longer("first", "abcde"), "abcdetsrifabcde");
Test.assertEquals(shorter_reverse_longer("hello", "bau"), "bauollehbau");
Test.assertEquals(shorter_reverse_longer("abcde", "fghi"), "fghiedcbafghi");

function test_shorter_reverse_longer(a,b){
  return (b.length > a.length) ? a+b.split("").reverse().join("")+a : b+a.split("").reverse().join("")+b
}

function random_string(){
  let l = Math.floor((Math.random() * 10));
  let text = "";
  let possible = "abcdefghijklmnopqrstuvwxyz";
  for( let i=0; i < l; i++ ) {
      text += possible.charAt(Math.floor(Math.random() * possible.length));
  }
  return text;
}

for(let i=0; i < 100; i++ ) {
  let a = random_string();
  let b = random_string();
  Test.assertEquals(test_shorter_reverse_longer(a,b),shorter_reverse_longer(a, b));
}